### DISCUSSION TITLE
g++-11: warning: could not understand version '13.00.00' ld: -rpath can only be used when targeting Mac OS X 10.5 or later

### DISCUSSION THREAD

**Discussion by lovi-bhatia** [Homebrew/discussions#3490] [Getting started]
I recently upgraded to macOS 13 public beta 2, and updated brew and gcc, but later I found this error and now I am unable to run any c++ code with gcc
g++-11: warning: could not understand version '13.00.00'
ld: -rpath can only be used when targeting Mac OS X 10.5 or later
collect2: error: ld returned 1 exit status
[Finished in 660ms with exit code 1]

**Comment by carlocab** [comment]
GCC does not support beta versions of macOS. You'll need to either fix GCC on your own or use macOS 10.15/11/12.

**Reply by lovi-bhatia** [reply]
how can I fix gcc?

**Reply by carlocab** [reply]
That, I don't know. It will require patches to work properly. The necessary patches may already be upstream, but they may also not be.

**Reply by lovi-bhatia** [reply]
I updated gcc-11 to gcc-12 now it's working fine.
Thanks

**Reply by Refat-Hassan** [reply]
> I updated gcc-11 to gcc-12 now it's working fine. Thanks

how to update gcc-11 to gcc-12 ?
please help me....
thanks a lot....

**Reply by Aarnav-JP** [reply]
how did you change your version from 11 to 12. I have installed gcc12 in my sys but when I run g++ -v it shows shows its still on gcc11.please help.

**Reply by Refat-Hassan** [reply]
> how did you change your version from 11 to 12. I have installed gcc12 in my sys but when I run g++ -v it shows shows its still on gcc11.please help.

if you install gcc 12 perfectly
you can try to check "gcc-12 -v"
then u can get see gcc-12 inshallah

**Reply by Aarnav-JP** [reply]
yeah that command is working for me indeed but when I compile my code using g++ file.cpp it is compiling via gcc11. if I do g++ -v its still on version 11 did you find anything to change it from 11 to 12 ?

**Reply by Refat-Hassan** [reply]
it's okay , your problem is c++ build system, first of all you need to create new build system where g++11 remove in this space rewrite g++12 ,,,, i think its enough for your problem .

**Reply by Aarnav-JP** [reply]
how can I do that please explain?thanks

**Comment by Refat-Hassan** [comment]
you can search youtube , how create c++ build system in sublime text on mac ...

**Comment by Anawrulkabir** [comment]
Previously i used run my c++ code with this sublime build.

{
  "cmd" : ["g++-10 $file_name -o $file_base_name && gtimeout 4s ./$file_base_name<inputf.in>outputf.in"], 
  "selector" : "source.c",
  "shell": true,
  "working_dir" : "$file_path"
}

source : https://github.com/luvk1412/Competitive-Programming/blob/master/C%2B%2B14_os_x.sublime-build

but currently is not working. Now i'm facing this problem below.


/bin/sh: g++-10: command not found
[Finished in 18ms with exit code 127]
[cmd: ['g++-10 map_practice.cpp -o map_practice && gtimeout 4s ./map_practice<inputf.in>outputf.in']]
[dir: /Users/anawrulkabirfahad/Desktop/competitive programming file]
[path: /Library/Frameworks/Python.framework/Versions/3.10/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:~/.dotnet/tools:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands]



I've already installed gcc-12 through Homebrew.
What can i do now.
Please Help me.

**Reply by Refat-Hassan** [reply]
You just change cmd line g++-10 to g++-12.
In simple word, Just replace 10 to 12

**Comment by kdevsolo** [comment]
remove the command line tools by the command
sudo rm -rf /Library/Developer/CommandLineTools

then go to this website and download the latest command line tools
https://developer.apple.com/download/all/

download latest gcc with homebrew that is : brew install gcc@12
then follow these steps
👉  cd /opt/homebrew/bin
👉  ln -s gcc-11 gcc
👉  ln -s g++-11 g++


In sublime then change g++-10 to g++-12 then it will work fine😃

**Reply by kdevsolo** [reply]
PS: create new build system for g++-12

**Reply by champagnealexandre** [reply]
thank you @keshavsaini2607 !

your solution worked for me, but I modified slightly the last 3 commands, so I used:

```
brew install gcc@12
cd /usr/local/bin
ln -sf gcc-12 gcc
ln -sf g++-12 g++
```

hope I haven’t broken anything overwriting the link in `/usr/local/bin` directly (instead of `/opt/homebrew/bin`, which didn’t exist on my system) but so far everything seems to be working fine!
